tests/nodes/supporting_nodes/test_group.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+import cript
+from cript.nodes.exceptions import UneditableAttributeError
+
+
+def test_create_group_from_json():
+    """
+    simply tests if a valid group node can be created from API JSON
+    """
+    pass
```

Comment:
I tried writing this test and the deserialization is coming out wrong. In the ` "admins":` and ` "users":` there is an extra 
` "groups": []`

any ideas, what is going wrong?


---

**Original group json:**
```json
{
    "node": "Group",
    "name": "my group name",
    "notes": "my group notes",
    "admins": [
        {
            "node": "User",
            "username": "my admin username",
            "email": "admin_email@email.com",
            "orcid": "0000-0000-0000-0001"
        }
    ],
    "users": [
        {
            "node": "User",
            "username": "my username",
            "email": "user@email.com",
            "orcid": "0000-0000-0000-0002"
        }
    ]
}
```

---

**Output group Json:**

```json
{
    "node": "Group",
    "name": "my group name",
    "admins": [
        {
            "node": "User",
            "username": "my admin username",
            "email": "admin_email@email.com",
            "orcid": "0000-0000-0000-0001",
            "groups": []
        }
    ],
    "users": [
        {
            "node": "User",
            "username": "my username",
            "email": "user@email.com",
            "orcid": "0000-0000-0000-0002",
            "groups": []
        }
    ],
    "notes": "my group notes"
}
```

Reply:
Yes, that is what the data model says.
Every user has a list of groups. Here they are empty, but still present.

This isn't good. This a potential cycle right there.
But the new serialization is correct, there should be a groups field there.

Reply:
Why the User serializer is not serializing Groups differently ? (just the minimal information)

Reply:
At the moment, we have not implemented to remove default values from the JSON.
It is certainly possible and not hard, just haven't been done yet.
I expect this to be resolved with #35

Reply:
BTW the official data model document does not specified that you can leave out default values from the JSON.

Reply:
This is an open problem we have.

We can't serialize node with their uuid, because the nodes may not exist on the back end, and hence have no uuid assigned to them.

This is an open problem.

Reply:
I see.

Reply:
I think the `groups` attribute within the `User` node was probably for the legacy API, so that Groups can point to the User and the User can point back at the Groups. I think we can remove them from the SDK fully actually. 
@dylanwal do you agree?